### PR TITLE
[#36] [iOS] [UI] As a user, I can see the Home screen with surveys

### LINKED ISSUE #36: [iOS] [UI] As a user, I can see the Home screen with surveys
# Why

The user should be able to see the Home screen with surveys provided by the API.

# Acceptance Criteria

- Show today and date label at the top of the screen.
- Show the user's avatar.
- Show the`>` button.
- Show a page view and page control
- Show survey detail, and it's descriptions at the bottom.
- Show the survey's thumbnail as a background image.

Image

[Figma](https://www.figma.com/file/GjRPOjDyZ6f4EDL3wKarRK/Challenge---Mobile-App?node-id=30-46&t=1xiBQA7GNv0kEx7j-0)

### PR DESCRIPTION
**Pull request by phongvhd93**
- Close #36

## What happened 👀

- Create UI for the Home screen
- Create a custom page indicator
- Add a `GestureVelocity` DynamicProperty to detect fast swipes on the home screen.

## Insight 📝

After a user login to the application, they will land on the home screen, which contains a collection of surveys.

## Proof Of Work 📹

https://user-images.githubusercontent.com/22606906/236748169-d3d1a267-ff30-453d-857e-cbaf08fcf4df.mp4

### CODE REVIEW

File: iosApp/PhongKMMIC/Sources/Presentation/Modules/Home/Views/HomeHeaderView.swift [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+struct HomeHeaderView: View {
+
+    var body: some View {
+        HStack {
+            VStack(spacing: 10.0) {
+                Text("MONDAY, JUNE 15")
+                    .foregroundColor(.white)
+                    .font(.boldSmall)
+                Text("TODAY")
```
Comment (markgravity) [inline iosApp/PhongKMMIC/Sources/Presentation/Modules/Home/Views/HomeHeaderView.swift]: We can move this `TODAY` text to `Localizable.strings` 🙋

File: iosApp/PhongKMMIC/Sources/Presentation/Modules/Home/Views/SurveyContentView.swift [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+                        print("Next button was tapped")
+                    } label: {
+                        R.image.rightChevron.image
+                            .frame(width: 56.0, height: 56.0)
+                            .background(Color.white)
+                            .clipShape(Circle())
+                    }
+                }
+            }
+            .padding([.horizontal], 20.0)
```
Comment (markgravity) [inline iosApp/PhongKMMIC/Sources/Presentation/Modules/Home/Views/SurveyContentView.swift]: We can simply `.padding(.horizontal, 20.0)` 🙏

File: iosApp/PhongKMMIC/Sources/Presentation/Modules/Home/Views/SurveyContentView.swift [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+            .frame(
+                width: geometryReader.size.width,
+                height: geometryReader.size.height,
+                alignment: .bottom
+            )
+            .background(content: {
+                if let image = uiModels[safe: currentPage]?.image {
```
Comment (markgravity) [inline iosApp/PhongKMMIC/Sources/Presentation/Modules/Home/Views/SurveyContentView.swift]: What's happened when the image is from an URL? 🤔

Comment (phongvhd93) [reply iosApp/PhongKMMIC/Sources/Presentation/Modules/Home/Views/SurveyContentView.swift]: I've already left an extension of `KingFisher` for that. That will be done in the integrated story. However, I'll update it with an actually url

Comment (phongvhd93) [reply iosApp/PhongKMMIC/Sources/Presentation/Modules/Home/Views/SurveyContentView.swift]: solved in 149e645

File: iosApp/PhongKMMIC/Sources/Presentation/Modules/Home/Views/SurveyContentView.swift (hunk 2 above)
Comment (phongvhd93) [reply iosApp/PhongKMMIC/Sources/Presentation/Modules/Home/Views/SurveyContentView.swift]: solved in 149e645

File: iosApp/PhongKMMIC/Sources/Presentation/Modules/Home/Views/HomeHeaderView.swift (hunk 1 above)
Comment (phongvhd93) [reply iosApp/PhongKMMIC/Sources/Presentation/Modules/Home/Views/HomeHeaderView.swift]: solved in 149e645

File: iosApp/PhongKMMIC/Sources/Presentation/Coordinator/Navigator.swift [hunk 4]
```diff
@@ -11,7 +11,7 @@ import SwiftUI
 
 final class Navigator: ObservableObject {
 
-    @Published var routes: Routes<Screen> = [.root(.splash, embedInNavigationView: true)]
+    @Published var routes: Routes<Screen> = [.root(.home, embedInNavigationView: true)]
```
Comment (markgravity) [inline iosApp/PhongKMMIC/Sources/Presentation/Coordinator/Navigator.swift]: Please revert this test code 🙏

File: iosApp/PhongKMMIC/Sources/Presentation/Modules/Home/Views/SurveyContentView.swift [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+    @State private var currentVisibility: Double = 1.0
+
+    @DragGestureVelocity private var velocity: CGVector
+
+    private var draggingSpeed: Double = 700.0
+    private var draggingLength: Double = 160.0
+    private var draggingVisibilityMultiplier: Double = 1.2
+
+    var body: some View {
+        // swiftlint:disable:next closure_body_length
```
Comment (markgravity) [inline iosApp/PhongKMMIC/Sources/Presentation/Modules/Home/Views/SurveyContentView.swift]: Please separate this body into small views, this rule is good to notify us when we need to chop this body into small ones 🙏

File: iosApp/PhongKMMIC/Sources/Supports/Extensions/Foundation/Collection+SafeIndex.swift [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+//  PhongKMMIC
+//
+//  Created by Phong Vo on 08/05/2023.
+//  Copyright © 2023 Nimble. All rights reserved.
+//
+
+import Foundation
+
+extension Collection {
+    subscript (safe index: Index) -> Element? {
```
Comment (markgravity) [inline iosApp/PhongKMMIC/Sources/Supports/Extensions/Foundation/Collection+SafeIndex.swift:13]: Missing break line here 🙏

File: iosApp/PhongKMMIC/Sources/Supports/Extensions/Foundation/Collection+SafeIndex.swift [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+        return indices.contains(index) ? self[index] : nil
```
Comment (markgravity) [inline iosApp/PhongKMMIC/Sources/Supports/Extensions/Foundation/Collection+SafeIndex.swift]: We can omit the `return` here 🙌

**Comment by phongvhd93** [comment]
@markgravity, I solved your comments above in the commit 246fad5

**Review by suho — APPROVED** [review]

File: iosApp/PhongKMMIC/Sources/Presentation/Coordinator/Navigator.swift (hunk 4 above)
Comment (markgravity) [reply iosApp/PhongKMMIC/Sources/Presentation/Coordinator/Navigator.swift]: @phongvhd93 You missed it 🙏

**Review by markgravity — APPROVED** [review]

**Pull request merged**